### ISSUE TITLE
Second sticky div on with tooltips/button: z-index issue

### ISSUE THREAD

**Issue by lotje-kinable** [livewire/flux#550]
Hi

Some of my pages have sticky div at the top of page. (client request)
But buttons with tooltips go above that sticky div
So I tried giving sticky div a z-1 tot z-9: and they still go above the sticky div.

Then I tried giving the sticky div a z-10, but now the sticky div goes above the sticky stashable sidebar...

![Image](https://github.com/user-attachments/assets/4199f039-1d50-4232-846b-a85c7007d2d4)

![Image](https://github.com/user-attachments/assets/77b71ed9-956b-4002-b4b9-60b176875b73)

The layout is the layout copied from the docs. 
Latest version of Flux and Livewire. 

### Route: 
```
Volt::route('/playground', 'flux-playground')->name('flux.playground');
```

### Minimal Volt component: 

```
<?php
use Livewire\Attributes\{Layout, Title};
use Livewire\Volt\Component;



new

class extends Component {


    public $options;

    public function mount()
    {

    $this->options = collect([
            (object)['id' => 1, 'name' => 'foo'],
            (object)['id' => 2, 'name' => 'bar'],
            (object)['id' => 3, 'name' => 'baz'],
            (object)['id' => 4, 'name' => 'qux'],
            (object)['id' => 5, 'name' => 'quux'],
            (object)['id' => 6, 'name' => 'corge'],
            (object)['id' => 7, 'name' => 'grault'],
            (object)['id' => 8, 'name' => 'garply'],
            (object)['id' => 9, 'name' => 'waldo'],
            (object)['id' => 10, 'name' => 'fred'],
            (object)['id' => 11, 'name' => 'plugh'],
            (object)['id' => 12, 'name' => 'xyzzy'],
            (object)['id' => 13, 'name' => 'thud'],
            (object)['id' => 14, 'name' => 'wibble'],
            (object)['id' => 15, 'name' => 'wobble'],
            (object)['id' => 16, 'name' => 'wubble'],
            (object)['id' => 17, 'name' => 'flob'],
            (object)['id' => 18, 'name' => 'flob'],
        ]);
    }

}; ?>

<div>

    <div class="sticky bg-pink-300 top-20">
        so this is sticky, with no z-index : icons/button go above this
    </div>
    <div class="sticky bg-yellow-300 z-9 top-28">
        sticky with z-9: icons/button go above this
    </div>
    <div class="sticky z-10 bg-green-300 top-32">
        sticky with z-10: visible when opening sidebar
    </div>
    <div>
        @foreach($options as $option)

            <p class="py-6" wire:key='option-{{ $option->id }}' value="{{ $option->id }}">
                <flux:tooltip content="Tooltip content">
                    <flux:button>{{ $option->name }}</flux:button>
                </flux:tooltip>
            </p>
        @endforeach
    </div>
</div>
```

Thanks in advance!

**Comment by jeffchown** [comment]
@lotje-kinable Have you tried giving the sidebar a `z-20` to 'counter' the `z-10` need?

**Comment by lotje-kinable** [comment]
> [@lotje-kinable](https://github.com/lotje-kinable) Have you tried giving the sidebar a `z-20` to 'counter' the `z-10` need?
I checked the sidebar blade and it has a z-20 in the original flux components.

**Comment by jeffchown** [comment]
@lotje-kinable Then, have you tried a higher z-index?  Just to see if it is possible to get it 'layered' the way you want? e.g. `z-50` or something?  Don't forget you may to use `!` with some classes to override the Flux default Tailwind utility classes, e.g. `!z-50`

**Comment by lotje-kinable** [comment]
@jeffchown the thing is: 
z-0 to z-9: the icon/button/tooltip go above the sticky div
>= z-10: icon/button/tooltip go under (as desired) but the sticky div is then on top of the sidebar when it is open (see screenshot above)
so if I use z-50 then it just stays on top of everything, but it has to stay under the sidebar. 

I am not sure that I have to change the default z-indexes of the flux components...

**Comment by jeffchown** [comment]
@lotje-kinable I'm not suggesting changing and leaving the z-index of the Flux sidebar, just testing a higher z-index on it as part of the investigation into the issue to determine if that (even temporarily) creates the desired result.

**Comment by jeffchown** [comment]
@lotje-kinable No worries :)

Interesting... Can you please provide your layout file and any other code that displays the sidebar, etc.?

**Comment by lotje-kinable** [comment]
@jeffchown 

Of course! 

I did have to set the standard flux backdrop to z-21 ( standard z-20), I posted an issue about this last week. And the only thing that worked was setting it to z-21 (one higher than the sidebar), so I thought, at first, that that might be the issue, but the green div still comes through and... I can't use the navigation, because the backdrop is disabling the whole page. So I had to leave it at z-21. 



### backdrop.blade.php (flux)

```
<div
    class="z-21 fixed inset-0 bg-black/10 hidden [[data-show-stashed-sidebar]_&]:block lg:[[data-show-stashed-sidebar]_&]:hidden"
    x-data x-on:click="document.body.removeAttribute('data-show-stashed-sidebar')"
></div>

```

### app.js

```
import './bootstrap';
import '../css/app.css';

import { Livewire, Alpine } from '../../vendor/livewire/livewire/dist/livewire.esm';
import Clipboard from '@ryangjchandler/alpine-clipboard'

Alpine.plugin(Clipboard)
```

### app.css: contains some @apply's but nothing with z-index, alpine.css holds the x-cloak css fix (from the alpinejs docs)
```
    @import 'components/alpine.css';

    @tailwind base;
    @tailwind components;
    @tailwind utilities;


```

### the layout file
in app.js the app.css is pulled in, along with 


```
<!DOCTYPE html>
<html lang="{{ str_replace('_', '-', app()->getLocale()) }}">

    <head>

        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="csrf-token" content="{{ csrf_token() }}">

        <title>{{ $title ?? 'TESTING FLUX' }}</title>



        @livewireStyles
        @vite(['resources/js/app.js'])

        @fluxStyles

    </head>
    <body class="min-h-screen bg-white dark:bg-zinc-800">

        <flux:header sticky class="border-b bg-zinc-50 dark:bg-zinc-900 border-zinc-200 dark:border-zinc-700">
            <flux:sidebar.toggle class="lg:hidden" icon="bars-2" />
            <flux:brand href="{{ route('welcome') }}" logo="https://fluxui.dev/img/demo/logo.png" name="TDW" class="max-lg:hidden dark:hidden" />

            <flux:brand href="#" logo="https://fluxui.dev/img/demo/dark-mode-logo.png" name="TDW" class="max-lg:!hidden hidden dark:flex" />
            <flux:navbar class="-mb-px max-lg:hidden">

                <flux:dropdown  class="max-lg:hidden">
                    <flux:navbar.item :current="request()->routeIs('flux.*')" icon-trailing="chevron-down">Flux</flux:navbar.item>
                    <flux:navmenu>

                        <flux:navmenu.item :href="route('flux.playground')">Flux playground</flux:navmenu.item>
                        <flux:navmenu.item :href="route('flux.foo')">foo</flux:navmenu.item>
                        <flux:navmenu.item :href="route('flux.bar')">bar</flux:navmenu.item>

                    </flux:navmenu>
                </flux:dropdown>
            </flux:navbar>
        </flux:header>


        <flux:sidebar sticky stashable class="border-r bg-zinc-50 lg:hidden dark:bg-zinc-700 border-zinc-200 dark:border-zinc-700">
            <flux:sidebar.toggle class="lg:hidden" icon="x-mark" />
            <flux:brand href="#" logo="https://fluxui.dev/img/demo/logo.png" name="Acme Inc." class="px-2 dark:hidden" />
            <flux:brand href="#" logo="https://fluxui.dev/img/demo/dark-mode-logo.png" name="Acme Inc." class="hidden px-2 dark:flex" />
            <flux:navlist variant="outline">

                    <flux:navlist.item :href="route('flux.playground')">Flux playground</flux:navlist.item>
                    <flux:navlist.item :href="route('flux.foo')">foo</flux:navlist.item>
                    <flux:navlist.item :href="route('flux.bar')">bar</flux:navlist.item>

            </flux:navlist>
        </flux:sidebar>


        <flux:main>
            {{ $slot }}
        </flux:main>

        <flux:toast />


        @stack('scripts')
        @stack('styles')

        @fluxScripts
        @livewireScriptConfig

    </body>

</html>
```

**Comment by jeffchown** [comment]
Thanks, @lotje-kinable .

I think this is one of those @calebporzio 's 'CSS is hard' type issues.

When playing with your code, I found out that the `flux:button`s are `relative`ly positioned.  That can cause unexpected results within the containing block element (see: https://stackoverflow.com/questions/8986071/how-to-use-z-index-with-relative-positioning).

What worked for me re: your issue, was to add `relative -z-10` to the `<div>` that wraps the content you want to appear beneath the sticky divs e.g.:

```
<div>
    <div class="sticky bg-pink-300 top-20">
        so this is sticky, with no z-index : icons/button go above this
    </div>
    <div class="sticky bg-yellow-300 z-9 top-28">
        sticky with z-9: icons/button go above this
    </div>
    <div class="sticky z-10 bg-green-300 top-32">
        sticky with z-10: visible when opening sidebar
    </div>

    <div class="relative -z-10">
        @foreach($options as $option)
            <p class="py-6" wire:key='option-{{ $option->id }}' value="{{ $option->id }}">
                <flux:tooltip content="Tooltip content">
                    <flux:button>{{ $option->name }}</flux:button>
                </flux:tooltip>
            </p>
        @endforeach
    </div>
</div>
```

**Comment by lotje-kinable** [comment]
@jeffchown okay, I'll try that righ now! 

I am working on my actual project too (code is from playground) and I just added some inputs with icons, and I can see the icon coming throught the navigation! I did not notice it before! 

![Image](https://github.com/user-attachments/assets/18a54170-df2a-45eb-a311-be0ce52598d9)
![Image](https://github.com/user-attachments/assets/894922d1-94a7-4218-bc47-d4df874733d2)

**Comment by lotje-kinable** [comment]
@jeffchown 'relative -z-10' on the div works! Thank you!!!

**Comment by jeffchown** [comment]
You're welcome, @lotje-kinable  - I learned something from this one too. 👍

**Comment by lotje-kinable** [comment]
@jeffchown 

I see the same thing with the icons, coming through the sidebar! 

so I put the 'relative -z-10' on the top div, and that fixes it as well!

![Image](https://github.com/user-attachments/assets/74f86ab2-e8fb-4ecb-ac0c-88c9c15e8583)

Thanks again, because I would have never though of using the 'inverse'!

**Comment by lotje-kinable** [comment]
@jeffchown okay so putting a relative -z-10 on the div around the flux:input, makes it unresposive to clicking, but clicking the label activates it! So weird

               <flux:field>

                    <flux:label>@if($form->is_booqit_ride) Totale prijs @else Ritprijs @endif</flux:label>

                    <div class="relative w-28 -z-10">

                        <flux:input wire:model='form.client_price' placeholder="0,00" type="number" icon="currency-euro" />

                    </div>

                    <flux:error name="form.client_price" />

                </flux:field>

**Comment by jeffchown** [comment]
@lotje-kinable I think it is one of those fixes you have to be careful not to over-use as it could cause unexpected results in other ways - like the ones you are now describing.

I wouldn't put `-z-10` on child elements, I would focus on using it, when absolutely necessary, on parent `<div>`s only.

This all seem to have started for you with the `sticky` elements the client requested.  Maybe this is one of those times where you have to consider an alternative that won't cause the initial issue.

**Comment by lotje-kinable** [comment]
@jeffchown since I am upgrading the code to Livewire 3 and using flux where possible, they are already used to the sticky div at the top, and it does makes sense to have it there. 
I can't use tables with pagination, because they want to see everything on one page (and as small as possible) (it is for taxi dispatching) and then they need to search for specific cars or drivers the search bar is always visible and sticky at the top... 

I need to do everything on this project ( since almost 3 years now). And using flux is making everything so much easier on the 'mainstream' pages, like settings and tables. But for the more specific pages, it's not plug and play, (that's normal), but I want to try and use as much flux as possible, because it's so much faster and it looks so good right away.

**Comment by jeffchown** [comment]
@lotje-kinable Sounds like you are making a lot of changes/updates at the same time.  Lots of moving parts.

I agree re: Flux making some thing soooo much easier.

Good luck!

**Comment by jeffchown** [comment]
@lotje-kinable Yes, Laravel + Livewire 3 + Alpine + Tailwind and now Flux are a great combination! - I guess we have to add an 'F' to 'TALL' = 'TALLF' ;)

You're welcome.

I'd close this issue.  People will still be able to find/search it if needed.

**Comment by lotje-kinable** [comment]
@jeffchown yeah... people should be aware of Flux! I would be so cool if the standard authentication views were flux! I think all that 
is necessary is available for free.

Have a nice day/night!

**Comment by lotje-kinable** [comment]
Update: 
Using the 'relative -z-10' on the div containing all the content: makes all buttons/input fields unclickable, unusable 
adding it to the icon or element: makes the button/element unclickable

So 'relative -z-10' is good for static content

**Comment by jeffchown** [comment]
Hi @lotje-kinable - Damn.  I thought we had this one figured out.  Just tried it and can confirm what you report about the buttons, etc.

**Comment by jeffchown** [comment]
@lotje-kinable I decided to try a fresh layout using the sidebar code from https://fluxui.dev/layouts/sidebar and putting your code from above in as the main content.

Everything for me worked as expected re: the sidebar.  I did not experience the issue shown in your 2nd (initial) screenshot.

I think we need to take a step back and look at your layout code as possibly the cause of the issue.

Can you post your full layout code so we can make sure we are replicating your exact situation?

**Comment by lotje-kinable** [comment]
@jeffchown 
I did a fresh copy too, just to make sure, and I have provided it (with a {{ $slot }} ) and a Volt component + a screenshot, where you can see the euro icon, through the header. 

Thank you for your patience!

![Image](https://github.com/user-attachments/assets/636e5601-1a7d-41a6-9b7b-4bab9238da82)


### Layout fresh copy from the link you added above.

```
<!DOCTYPE html>
<html lang="{{ str_replace('_', '-', app()->getLocale()) }}">

    <head>

        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="csrf-token" content="{{ csrf_token() }}">

        <title>{{ $title ?? 'TESTING FLUX' }}</title>

        @livewireStyles
        @vite(['resources/js/app.js'])

        @fluxStyles

    </head>
    <body class="min-h-screen bg-white dark:bg-zinc-800">


        <flux:sidebar sticky stashable class="border-r bg-zinc-50 dark:bg-zinc-900 border-zinc-200 dark:border-zinc-700">
            <flux:sidebar.toggle class="lg:hidden" icon="x-mark" />

            <flux:brand href="#" logo="https://fluxui.dev/img/demo/logo.png" name="Acme Inc." class="px-2 dark:hidden" />
            <flux:brand href="#" logo="https://fluxui.dev/img/demo/dark-mode-logo.png" name="Acme Inc." class="hidden px-2 dark:flex" />

            <flux:input as="button" variant="filled" placeholder="Search..." icon="magnifying-glass" />

            <flux:navlist variant="outline">
                <flux:navlist.item icon="home" href="#" current>Home</flux:navlist.item>
                <flux:navlist.item icon="inbox" badge="12" href="#">Inbox</flux:navlist.item>
                <flux:navlist.item icon="document-text" href="#">Documents</flux:navlist.item>
                <flux:navlist.item icon="calendar" href="#">Calendar</flux:navlist.item>

                <flux:navlist.group expandable heading="Favorites" class="hidden lg:grid">
                    <flux:navlist.item href="#">Marketing site</flux:navlist.item>
                    <flux:navlist.item href="#">Android app</flux:navlist.item>
                    <flux:navlist.item href="#">Brand guidelines</flux:navlist.item>
                </flux:navlist.group>
            </flux:navlist>

            <flux:spacer />

            <flux:navlist variant="outline">
                <flux:navlist.item icon="cog-6-tooth" href="#">Settings</flux:navlist.item>
                <flux:navlist.item icon="information-circle" href="#">Help</flux:navlist.item>
            </flux:navlist>

            <flux:dropdown position="top" align="start" class="max-lg:hidden">
                <flux:profile avatar="https://fluxui.dev/img/demo/user.png" name="Olivia Martin" />

                <flux:menu>
                    <flux:menu.radio.group>
                        <flux:menu.radio checked>Olivia Martin</flux:menu.radio>
                        <flux:menu.radio>Truly Delta</flux:menu.radio>
                    </flux:menu.radio.group>

                    <flux:menu.separator />

                    <flux:menu.item icon="arrow-right-start-on-rectangle">Logout</flux:menu.item>
                </flux:menu>
            </flux:dropdown>
        </flux:sidebar>

        <flux:header sticky class="!block bg-white lg:bg-zinc-50 dark:bg-zinc-900 border-b border-zinc-200 dark:border-zinc-700">
            <flux:navbar class="w-full lg:hidden">
                <flux:sidebar.toggle class="lg:hidden" icon="bars-2" inset="left" />

                <flux:spacer />

                <flux:dropdown position="top" align="start">
                    <flux:profile avatar="https://fluxui.dev/img/demo/user.png" />

                    <flux:menu>
                        <flux:menu.radio.group>
                            <flux:menu.radio checked>Olivia Martin</flux:menu.radio>
                            <flux:menu.radio>Truly Delta</flux:menu.radio>
                        </flux:menu.radio.group>

                        <flux:menu.separator />

                        <flux:menu.item icon="arrow-right-start-on-rectangle">Logout</flux:menu.item>
                    </flux:menu>
                </flux:dropdown>
            </flux:navbar>

            <flux:navbar scrollable>
                <flux:navbar.item href="#" current>Dashboard</flux:navbar.item>
                <flux:navbar.item badge="32" href="#">Orders</flux:navbar.item>
                <flux:navbar.item href="#">Catalog</flux:navbar.item>
                <flux:navbar.item href="#">Configuration</flux:navbar.item>
            </flux:navbar>
        </flux:header>


            <flux:main>
                <flux:heading size="xl" level="1">Good afternoon, Olivia</flux:heading>

                <flux:subheading size="lg" class="mb-6">Here's what's new today</flux:subheading>

                <flux:separator variant="subtle" />


                {{ $slot }}


            </flux:main>



        <flux:toast />

        @fluxScripts
        @livewireScriptConfig

    </body>

</html>
```

### Volt component 


```
<?php
use Livewire\Attributes\{Layout, Title};
use Livewire\Volt\Component;



new

class extends Component {


    public $options;

    public function mount()
    {

    $this->options = collect([
            (object)['id' => 1, 'name' => 'foo', 'price' => '10,0'],
            (object)['id' => 2, 'name' => 'bar', 'price' => '20,0'],
            (object)['id' => 3, 'name' => 'baz', 'price' => '30,0'],
            (object)['id' => 4, 'name' => 'qux', 'price' => '40,0'],
            (object)['id' => 5, 'name' => 'quux', 'price' => '50,0'],
            (object)['id' => 6, 'name' => 'corge', 'price' => '60,0'],
            (object)['id' => 7, 'name' => 'grault', 'price' => '70,0'],
            (object)['id' => 8, 'name' => 'garply', 'price' => '80,0'],
            (object)['id' => 9, 'name' => 'waldo', 'price' => '90,0'],
            (object)['id' => 10, 'name' => 'fred', 'price' => '1,0'],
            (object)['id' => 11, 'name' => 'plugh', 'price' => '2,0'],
            (object)['id' => 12, 'name' => 'xyzzy', 'price' => '3,0'],
            (object)['id' => 13, 'name' => 'thud', 'price' => '4,0'],

        ]);
    }


}; ?>

<div>


    <div>
        @foreach($options as $option)

            <p class="py-6" wire:key='option-{{ $option->id }}' value="{{ $option->id }}">
                <flux:tooltip content="Tooltip content">
                    <flux:button>{{ $option->name }}</flux:button>
                </flux:tooltip>


                <flux:field>

                    <flux:label>Price in euro</flux:label>

                    <div class="w-28">

                        <flux:input :value="$option->price" type="text" icon="currency-euro" />

                    </div>



                </flux:field>
            </p>
        @endforeach
    </div>
</div>
```

**Comment by jeffchown** [comment]
@lotje-kinable I tried your latest code.

The initial issue from this thread no longer seems to exist - the buttons, etc. properly go 'under' the sticky div and the navbar and the sticky div does not bleed through the responsive sidebar (using this code for your component):

```
<div>
    <div class="sticky z-10 bg-green-300 top-32">
        sticky with z-10: visible when opening sidebar
    </div>

    @foreach($options as $option)

        <p class="py-6" wire:key='option-{{ $option->id }}' value="{{ $option->id }}">
            <flux:tooltip content="Tooltip content">
                <flux:button>{{ $option->name }}</flux:button>
            </flux:tooltip>


            <flux:field>
                <flux:label>Price in euro</flux:label>

                <div class="w-28">
                    <flux:input :value="$option->price" type="text" icon="currency-euro" class="" />
                </div>
            </flux:field>
        </p>
    @endforeach
</div>
```

As for the icon bleeding through the navbar, that issue I can recreate.

As the main issue is this thread is resolved and this thread is getting cluttered and a little confusing, I think it would be best to close this issue and start a new issue **just** about the icon issue.

I believe you can create much simpler example code for the icon issue that may get you an answer more quickly.

**Comment by lotje-kinable** [comment]
@jeffchown that does fix the header!!  But not the sidebar.. see screenshot (I did a view:clear and build)

![Image](https://github.com/user-attachments/assets/a1e436b8-e0dc-4169-9779-4988eb172c6e)

**Comment by lotje-kinable** [comment]
@jeffchown sidenote:  this is a screenshot from my exisiting application, (and I put the z-20 on both that header and sidebar,) and what worries me is that the buttons (x-marks) are still clickable while the sidebar is open and you can also see them in white, not greyed out like the rest of the fields. 

![Image](https://github.com/user-attachments/assets/45aa9205-f631-4783-b257-5c73ee05cfb8)

**Comment by jeffchown** [comment]
Hi @lotje-kinable - you are adding issues to this thread that are different than the original issue (while they may seem related)

It is making the thread very cluttered and hard to follow/help.

When debugging something, it is easier to tackle each issue on its own (and in its own issue in GitHub).

I would recommend posting your latest comment re: buttons in a new issue so people can focus on that specific issue.

**Comment by lotje-kinable** [comment]
@jeffchown okay

**Comment by jeffchown** [comment]
@lotje-kinable I don't see a need to delete anything.

**Comment by calebporzio** [comment]
Thanks for the report @lotje-kinable - however, after pulling in your code and failing to easily reproduce your screenshot I gave up. There is too much unrelated code in these snippets and the problem is not described poignantly enough. I'm sorry to close this, but please make these issues as clear and concise as you possibly can before we invest time in solving them.

(Thanks for being so helpful @jeffchown - you're a legend)

**Comment by lotje-kinable** [comment]
@jeffchown and everyone: I found the solution: I must have set "important: true", in my tailwind.config.js and that's what caused this!
